packages/block-library/src/column/edit.js

```diff
@@ -113,44 +102,29 @@ export default compose(
 					'core/block-editor'
 				);
 
-				// Constrain or expand siblings to account for gain or loss of
-				// total columns area.
-				const columns = getBlocks( getBlockRootClientId( clientId ) );
-				const adjacentColumns = getAdjacentBlocks( columns, clientId );
+				// Update column width.
+				updateBlockAttributes( clientId, { width } );
 
-				// The occupied width is calculated as the sum of the new width
-				// and the total width of blocks _not_ in the adjacent set.
-				const occupiedWidth =
-					width +
-					getTotalColumnsWidth(
-						difference( columns, [
-							find( columns, { clientId } ),
-							...adjacentColumns,
-						] )
-					);
+				const rootClientId = getBlockRootClientId( clientId );
+				const columns = getBlocks( rootClientId );
 
-				// Compute _all_ next column widths, in case the updated column
-				// is in the middle of a set of columns which don't yet have
-				// any explicit widths assigned (include updates to those not
-				// part of the adjacent blocks).
-				const nextColumnWidths = {
-					...getColumnWidths( columns, columns.length ),
-					[ clientId ]: toWidthPrecision( width ),
-					...getRedistributedColumnWidths(
-						adjacentColumns,
-						100 - occupiedWidth,
-						columns.length
-					),
-				};
+				// Check if sum of all column widths exceeds 100.
```

Comment:
It was, but in response to the issues @talldan raised with how the percentages were working, I have since changed to `fr` units. They are being used like percentage units though, so the `100` still makes sense here, and I added some (hopefully) explanatory comments.